File: ios/Banner/YandexBannerViewManager.m
#if !TARGET_OS_MACCATALYST

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_REMAP_MODULE(BannerView, YandexBannerViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(adUnitId, NSString)
RCT_EXPORT_VIEW_PROPERTY(adSize, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(adRequest, NSDictionary)

RCT_EXPORT_VIEW_PROPERTY(onAdLoaded, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onAdFailedToLoad, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onAdImpression, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onAdClicked, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onLeftApplication, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onAdClose, RCTDirectEventBlock)

@end

#endif

File: ios/Banner/YandexBannerViewManager.swift
#if !targetEnvironment(macCatalyst)

import Foundation
import React

@objc(YandexBannerViewManager)
class YandexBannerViewManager: RCTViewManager {
    private let bannerAdSizeFactory: TypedBannerAdSizeFactory
    
    convenience override init() {
        let bannerAdSizeFactory = TypedBannerAdSizeFactory()
        self.init(bannerAdSizeFactory: bannerAdSizeFactory)
    }
    
    init(bannerAdSizeFactory: TypedBannerAdSizeFactory) {
        self.bannerAdSizeFactory = bannerAdSizeFactory
    }
    
    override func view() -> UIView {
        YandexBannerView()
    }
    
    override static func requiresMainQueueSetup() -> Bool {
        true
    }
}

#endif
